pkg/controller/node/controller_utils.go

```diff
@@ -77,9 +86,28 @@ func deletePods(kubeClient clientset.Interface, recorder record.EventRecorder, n
 		}
 		remaining = true
 	}
+	
+	if len(updateErrList) > 0 {
+		return false, utilerrors.NewAggregate(updateErrList)
+	}
 	return remaining, nil
 }
 
+// setPodTerminationReason attempts to set a reason and message in the pod status, updates it in the apiserver,
+// and returns an error if it encounters one.
+func setPodTerminationReason(kubeClient clientset.Interface, pod *api.Pod, nodeName string) (*api.Pod, error) {
+	updatedPod := pod
+	var err error
+	if pod.Status.Reason != node.NodeUnreachablePodReason {
+		pod.Status.Reason = node.NodeUnreachablePodReason
+		pod.Status.Message = fmt.Sprintf(node.NodeUnreachablePodMessage, nodeName, pod.Name)
+		if updatedPod, err = kubeClient.Core().Pods(pod.Namespace).UpdateStatus(pod); err != nil {
```

Comment:
From my tests, when the kubelet comes back, it removes the reason set here as it updates pod status, but if we are here, the deletionTimeStamp is already set, so, it just moves towards actual deletion of the pod. From kubectl get's standpoint, the pods transition from `Unknown` to `Terminating` before getting deleted.